pkg/config/remote/client.go

```diff
@@ -152,6 +157,18 @@ func newClient(agentName string, updater ConfigUpdater, doTufVerification bool,
 	backoffPolicy := backoff.NewPolicy(minBackoffFactor, pollInterval.Seconds(),
 		maximalMaxBackoffTime.Seconds(), recoveryInterval, false)
 
+	// If we're the cluster agent, we want to report our cluster name and cluster ID in order to allow products
+	// relying on remote config to identify this RC client to be able to write predicates for config routing
+	clusterName := ""
+	if flavor.GetFlavor() == flavor.ClusterAgent {
+		hname, err := hostname.Get(context.TODO())
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** Let's use another context that `.TODO()`. We can either use `context.Background()` or take the context that is defined line 172.‏

Reply:
This works for me, although neither `Background()` nor the context in line 172 changes anything about the functional behavior here.

I went with TODO() on account that the docs state that `when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a context parameter)`.

The proper way to address this is that, allowing the caller to set a timeout for these operations or cancel us if we hang.

Reply:
I agree with what you've written, though here it feels like as this is an intializing function, background should be used. Besides, as we define a context right after, it only feels right to use it.

It's not a big deal, and, as you said, the functional behavior doesn't change, hence the nitpick 😄 

I don't have a strong opinion on this to be very honest, so it's as you wish

Reply:
> Besides, as we define a context right after, it only feels right to use it.

That context is so that owners of the Client can have the underlying processing goroutine gracefully shutdown if needed. At this point in time only we can call the cancel func for the hostname and cluster-name functions, and we have no intention of doing so. So it's not really right, or wrong for that matter, to use it. 😁 

We could define our own Background() in this func and layer a timeout on it for those two funcs, but we aren't. The documentation about initialization is mostly just to say "hey, here's how you should start a context chain" since the context API is immutable and also buried behind an interface. Background() by itself though means and does nothing special.

FWIW in the `context` package, TODO() and Background() are both defined as `new(emptyCtx)`. TODO() just looks distinct as a signal to developers to wire this up to a passed on Context down the road. One could argue that's the only correct thing to do here so that the main agent initialization can cancel or time us out of those functions if it wants to. I'm going to do that - it's an easy change and it's the truly right thing to do.